ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/metrics/DeltaFilesMetricReporter.java

```diff
@@ -206,7 +216,7 @@ public static void backPropagateAcidMetrics(JobConf jobConf, Configuration conf)
   }
 
   public static void close() {
-    if (getInstance() != null) {
+    if (getInstance() != null && getInstance().acidMetricsExtEnabled) {
```

Comment:
I don't think this is necessary... if the instance exists it should be shut down, that's all...

Reply:
the executorService is created in the configure method. So if it was not called this will throw a NPE if I understand.

Reply:
Oh, then probably null checking the executorService would be best here